pkg/iac-providers/cft/v1/load-dir_test.go

```diff
@@ -107,23 +113,48 @@ func TestLoadIacDir(t *testing.T) {
 
 func TestCFTMapper(t *testing.T) {
 	root := filepath.Join(basePath, provider)
-	dirList := make([]string, 0)
-	err := filepath.Walk(root, func(filePath string, fileInfo os.FileInfo, err error) error {
-		if fileInfo != nil && fileInfo.IsDir() {
-			dirList = append(dirList, filePath)
-		}
-		return err
-	})
-
+	dirList, err := ioutil.ReadDir(root)
 	if err != nil {
-		t.Error(err)
+		log.Fatal(err)
 	}
 
 	cftv1 := CFTV1{}
-	for i := 1; i < len(dirList); i++ {
-		dir := dirList[i]
-		t.Run(dir, func(t *testing.T) {
-			_, gotErr := cftv1.LoadIacDir(dir, false)
+	for _, dir := range dirList {
+		resourceDir := filepath.Join(root, dir.Name())
+		t.Run(resourceDir, func(t *testing.T) {
+			allResourceConfigs, gotErr := cftv1.LoadIacDir(resourceDir, false)
+
+			// load expected output.json from test artifacts
+			var testArc output.AllResourceConfigs
+			outputData, err := ioutil.ReadFile(filepath.Join(resourceDir, "output.json"))
+			if err != nil {
+				t.Errorf("error reading output.json ResourceConfig, %T", err)
+			}
+
+			err = json.Unmarshal(outputData, &testArc)
+			if err != nil {
+				t.Errorf("error loading output.json ResourceConfig, %T", err)
+			}
+
+			// check if resourcetype and resources are present
+			for name, resources := range testArc {
+				if allResourceConfigs[name] == nil {
+					t.Errorf("resource Type %s from test data %s not found", name, resourceDir)
+				}
+				for _, testResource := range resources {
```

Comment:
Hey @sigmabaryon, I was wondering if `deepEqual` would have worked here? Because we are validating if the `ID` is present or not, but not validating other fields of the config resource.